tests/integration.rs

```diff
@@ -6293,6 +6297,292 @@ mod common_parallel {
+                    Some(std::net::Ipv4Addr::from_str(&guest.network.host_ip).unwrap()),
+                    None,
+                    &mut None,
+                    None,
+                    num_queue_pairs,
+                    Some(libc::O_RDWR | libc::O_NONBLOCK),
+                )
+                .unwrap();
+                let net_params = format!(
+                    "id={},fd=[{},{}],mac={},ip={},mask=255.255.255.0,num_queues={}",
+                    net_id,
+                    taps[0].as_raw_fd(),
+                    taps[1].as_raw_fd(),
+                    guest.network.guest_mac,
+                    guest.network.host_ip,
+                    num_queue_pairs * 2
+                );
+
+                assert!(remote_command(
+                    &api_socket_source,
+                    "add-net",
+                    Some(net_params.as_str()),
+                ));
+                thread::sleep(std::time::Duration::new(10, 0));
+                // close the fds once guest boots up, as CH duplicates them before using
+                for tap in taps.iter() {
+                    unsafe { libc::close(tap.as_raw_fd()) };
+                }
+            }
+
+            snapshot_and_check_events(&api_socket_source, &snapshot_dir, &event_path);
+        });
+
+        // Shutdown the source VM and check console output
+        let _ = child.kill();
+        let output = child.wait_with_output().unwrap();
+        handle_child_output(r, &output);
+
+        let r = std::panic::catch_unwind(|| {
+            assert!(String::from_utf8_lossy(&output.stdout).contains(&console_text));
+        });
+
+        handle_child_output(r, &output);
+
+        // Remove the vsock socket file.
+        Command::new("rm")
+            .arg("-f")
+            .arg(socket.as_str())
+            .output()
+            .unwrap();
+
+        let api_socket_restored = format!("{}.2", temp_api_path(&guest.tmp_dir));
+        let event_path_restored = format!("{}.2", temp_event_monitor_path(&guest.tmp_dir));
+
+        // Restore the VM from the snapshot
+        let mut child = GuestCommand::new(&guest)
+            .args(["--api-socket", &api_socket_restored])
+            .args([
+                "--event-monitor",
+                format!("path={event_path_restored}").as_str(),
+            ])
+            .capture_output()
+            .spawn()
+            .unwrap();
+        thread::sleep(std::time::Duration::new(2, 0));
+
+        let taps = net_util::open_tap(
+            Some(tap_name),
+            Some(std::net::Ipv4Addr::from_str(&guest.network.host_ip).unwrap()),
+            None,
+            &mut None,
+            None,
+            num_queue_pairs,
+            Some(libc::O_RDWR | libc::O_NONBLOCK),
+        )
+        .unwrap();
+        let restore_params = format!(
+            "source_url=file://{},net_fds=[{}@[{},{}]]",
+            snapshot_dir,
+            net_id,
+            taps[0].as_raw_fd(),
```

Comment:
Looks like you don't test the `ch-remote` / API version - only the restore via command line?

Reply:
The test uses `ch-remote` for restore operation. (See Line: 6495 https://github.com/cloud-hypervisor/cloud-hypervisor/pull/6402/files#diff-eb771f54a1fdc1c330a82981d7d27d1202a5cbe3a59581ff67af5e3361f5f90eR6495)

Reply:
Apologies - my mistake! Our integration tests are really hard to read!